guide: reorder how-tos and copy edits; expand on adding existing dependencies to a stage (#1914)
iterative/dvc.org: content/docs/command-reference/run.md

  also means that the stage command needs to recreate any directory structures
  defined as outputs every time its executed by DVC.

> - In some situations, we have previously executed a stage, and later notice that

Reviewer: situations , ... , and later , ... that something/something used by, or .... 

and only after that 

are missing 

this is very hard to read still. to many clauses per sentence.

in most cases in this PR it should be significantly simplified still.

Author: It should be easier to read after it gets generalized per https://github.com/iterative/dvc.org/pull/1914#discussion_r528391826 above.

Author: Just rewrote this and did other updates to how-tos per some recent discussions around `remove` on Discord. PTAL.

>   some of the files/directories are missing from the `deps` or `outs` fields of

Reviewer: do we need to mention `outs` and `deps` fields here? 

also consider reverse order of the sentences.

Author: mentioning deps/outs fields was just a shorter alternative to "files/directories used by the stage as dependencies or created as outputs are missing from". BTW @imhardikj how we have different text in commit vs here in run. We have to pick one 😅

Author: BTW @imhardikj how we have different text in commit vs here in run. We have to pick one 😅

Reviewer 2: Updated. Similar now.

  defined as outputs every time its executed by DVC.

- In some situations we have executed a stage and later notice that some of the

>   existing dependencies, or output files/directories it creates, which are

Author: Same, deps are not created by stages (and unnecessary comma).

>   existing dependencies of the stage or output files/directories it creates,

Author: Same. Deps are also files/dirs. Like this is sounds as if only outs are files/dirs.

Reviewer 2: Also done

>   files/directories used by the stage as dependencies, or created as outputs,

Reviewer: same here - too many details, hard to read - sacrifice some:

`which are already in workspace`
`deps and outs field`
?

Author: Please apply my suggestion from commit here too @imhardikj.

Author: > deps and outs field

This is a good idea. Something like:

```suggestion
  files/directories are missing from the `deps` or `outs` fields of `dvc.yaml`.
```

It assumes the user will know what it means to be in those fields but I guess you can always click on `dvc.yaml`, find the fields, and see the "dependency" and "output" tooltips from there.

Reviewer 2: Updated this

Reviewer: It's better, @imhardikj ! It's always better to go first with a simple statement sentence - what it does. Then explanation if it's needed. This still a very complicates sentence structure and wrong order of sentences within a paragraph to my mind.

Reviewer: outdate comment ^^, disregrad

>   outputs, which are already in the workspace, are missing from `dvc.yaml`

Author: Same

Reviewer 2: Done
